Build-Installer.ps1

```diff
@@ -72,6 +72,45 @@ function GetConstraintFile() {
     return "espidf.constraints.v${ShortVersion}.txt"
 }
 
+# Extract major and minor version from a version string and return as [System.Version] object
+# For example, "v5.0.1" returns [System.Version]"5.0", "v4.4.2-rc" returns [System.Version]"4.4"
+# Handles patch versions correctly by extracting only major.minor parts
+function GetVersionObject() {
+    param (
+        [Parameter(Mandatory)]
+        [String]
+        $VersionString
+    )
+    
+    # Remove 'v' prefix and any suffix after '-'
+    $CleanVersion = $VersionString -replace "^v" -replace "-.*$"
+    $SplitVersion = $CleanVersion -split "\."
+    
+    # Get major and minor parts, default to 0 if missing
+    $Major = if ($SplitVersion.Length -gt 0 -and $SplitVersion[0] -match '^\d+$') { [int]$SplitVersion[0] } else { 0 }
+    $Minor = if ($SplitVersion.Length -gt 1 -and $SplitVersion[1] -match '^\d+$') { [int]$SplitVersion[1] } else { 0 }
+    
+    return [System.Version]"${Major}.${Minor}"
+}
+
+# Compare two version strings properly
+# Returns $true if Version1 >= Version2, $false otherwise
+function CompareVersions() {
```

Comment:
Nitpick, as this repo will be deprecated soon:
I would suggest more self-explanatory naming method like `Test-Ver1GreaterOrEqualVer2` or `Is-Version2AtLeastVersion1` for easier code readability.